smack-experimental/src/main/java/org/jivesoftware/smackx/blocking/BlockingCommandManager.java

```diff
@@ -0,0 +1,124 @@
+/**
+ *
+ * Copyright 2016 Fernando Ramirez
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.blocking;
+
+import java.util.List;
+
+import org.jivesoftware.smack.SmackException.NoResponseException;
+import org.jivesoftware.smack.SmackException.NotConnectedException;
+import org.jivesoftware.smack.XMPPConnection;
+import org.jivesoftware.smack.XMPPException.XMPPErrorException;
+import org.jivesoftware.smack.packet.IQ;
+import org.jivesoftware.smackx.blocking.element.BlockContactIQ;
+import org.jivesoftware.smackx.blocking.element.BlockListIQ;
+import org.jivesoftware.smackx.blocking.element.BlockingCommandElements;
+import org.jivesoftware.smackx.blocking.element.GetBlockListIQ;
+import org.jivesoftware.smackx.blocking.element.UnblockContactIQ;
+import org.jivesoftware.smackx.disco.ServiceDiscoveryManager;
+import org.jxmpp.jid.Jid;
+
+/**
+ * Blocking command manager class.
+ * 
+ * @author Fernando Ramirez
+ * @see <a href="http://xmpp.org/extensions/xep-0191.html">XEP-0191: Blocking
+ *      Command</a>
+ */
+public final class BlockingCommandManager {
+
+    /**
+     * Returns true if Blocking Command is supported by the server.
+     * 
+     * @param connection
+     * @return true if Blocking Command is supported by the server.
+     * @throws NoResponseException
+     * @throws XMPPErrorException
+     * @throws NotConnectedException
+     * @throws InterruptedException
+     */
+    public static boolean isSupportedByServer(XMPPConnection connection)
+            throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException {
+        return ServiceDiscoveryManager.getInstanceFor(connection)
+                .serverSupportsFeature(BlockingCommandElements.NAMESPACE);
+    }
+
+    /**
+     * Returns the blocking list.
+     * 
+     * @param connection
+     * @return the blocking list
+     * @throws NoResponseException
+     * @throws XMPPErrorException
+     * @throws NotConnectedException
+     * @throws InterruptedException
+     */
+    public static List<Jid> getBlockingList(XMPPConnection connection)
+            throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException {
+        GetBlockListIQ getBlockListIQ = new GetBlockListIQ();
+        IQ responseIq = connection.createPacketCollectorAndSend(getBlockListIQ).nextResultOrThrow();
+        BlockListIQ blockListIQ = (BlockListIQ) responseIq;
```

Comment:
FYI: You don't need this extra line. nextResultOrThrow() is generic and does the cast for you. But you don't need to change it if you don't want to. Just wanted to point this out.